Automattic/wp-calypso
# Button component: remove wpcom specific resets

Shuffle some resets around. Button component contains some wpcom specific CSS overrides that shouldn't be in the component CSS.

#### Changes proposed in this Pull Request

* Remove `-webkit-appearance` rule since it's already generated by Autoprefixer when having `appearance` for `input`
* Remove `.layout__content` and `.dialog__content` specific resets from Button component and place them directly to those components.

#### Testing instructions

* Confirm that `type=['reset']` buttons work
* Confirm that buttons across the app work and look like they used to
